src/CodeGen_Hexagon.cpp

```diff
@@ -1080,23 +1080,45 @@ Value *CodeGen_Hexagon::shuffle_vectors(Value *a, Value *b,
     // TODO: There are more HVX permute instructions that could be
     // implemented here, such as vdelta/vrdelta.
 
-    if (element_bits <= 16) {
-        return vlut(concat_vectors({a, b}), indices);
-    } else {
-        return CodeGen_Posix::shuffle_vectors(a, b, indices);
-    }
+    return vlut(concat_vectors({a, b}), indices);
 }
 
-Value *CodeGen_Hexagon::vlut(Value *lut, Value *idx, int min_index, int max_index) {
+Value *CodeGen_Hexagon::vlut256(Value *lut, Value *idx, int min_index, int max_index) {
     bool is_128B = target.has_feature(Halide::Target::HVX_128);
     llvm::Type *lut_ty = lut->getType();
+    llvm::Type *result_ty = llvm::VectorType::get(lut_ty->getVectorElementType(),
+                                   idx->getType()->getVectorNumElements());
+
+    int replicate = lut_ty->getScalarSizeInBits()/16;
+    int lut_max = (replicate > 2) ? 256/replicate : 256;
```

Comment:
Why does this not go down for replicate = 2?

I'd expect this to need to be:

    lut_max = 256 / replicate

Also, I think there is a way to do this without this reduced lut_max requirement. If we deinterleaved the LUT instead of interleaving the indices, we wouldn't have this constraint. But this approach might have more overhead? Either way, perhaps leave it as a TODO.